src/Common/OptimizedRegularExpression.cpp

```diff
@@ -14,13 +14,42 @@ namespace DB
     }
 }
 
+namespace
+{
 
-template <bool thread_safe>
-void OptimizedRegularExpressionImpl<thread_safe>::analyze(
+struct Literal
+{
+    std::string literal;
```

Comment:
Just speculating: A literal is part of a regexp which exists independently. Could we maybe store a std::string_view instead of a std::string (l. 22)?

Reply:
yes, but analyze is not frequently called so i think it's ok for now